### PR TITLE
Add CM500 profile

### PR DESCRIPTION
**Pull request by redawl**
Hi, I have tested this on my CM500 modem, and am able to dump ram in with patched dump. 

I think this is ready to be merged, but I will continue hacking on this profile. 

My main hurdle right now is the bootloader on the CM500 is missing the 'p' command, so bcm2-utils cannot read the memory map. 

Any ideas on how to move forward would be great! I have identified SpiFlashRead and SpiFlashWrite, but am unable to use them so far.

### CODE REVIEW

File: profile.h [hunk 1]
```diff
@@ -103,6 +103,7 @@ enum bcm2_arch
 	BCM2_3383 = 0x3383,
 	BCM2_3384 = 0x3384,
 	BCM2_3390 = 0x3390,
+	BCM2_3843 = 0x3843,
```
Comment (jclehner) [inline profile.h:106]: Please keep this sorted (i.e. put it before `BCM2_3345`)

Comment (jclehner) [reply profile.h:106]: I just noticed it's `3843`, not `3343`. However, all Broadcom CM chipsets start with `33`. According to [deviwiki](https://deviwiki.com/wiki/Netgear_CM500) this should be `33843`!

**Comment by jclehner** [comment]
> Any ideas on how to move forward would be great! I have identified SpiFlashRead and SpiFlashWrite, but am unable to use them so far. 

What flash chip does this modem use? Can you post the full boot log from the serial console?

**Comment by redawl** [comment]
[screenlog.txt](https://github.com/jclehner/bcm2-utils/files/14932587/screenlog.txt)
[bootloader2.txt](https://github.com/jclehner/bcm2-utils/files/14932591/bootloader2.txt)

Interesting that deviwiki says 3384, while the bootlog shows 3843. Maybe a typo in the firmware? I will fix :)

flashchip is [25Q64FVAIG](https://pdf1.alldatasheet.com/datasheet-pdf/view/1156000/WINBOND/25Q64FVAIG.html)

**Comment by redawl** [comment]
I have tried dumping the flash chip with bus pirate, but the modem boots when I power VCC, and interferes with read operations. I do not have a soldering kit to try lifting the VCC pin, but when I get one I will try that.

**Comment by redawl** [comment]
Any updates on getting this merged?

In the meantime I have been able to dump the flash chip with  my bus pirate, it turned out the bp firmware I had was bad, so an older firmware version allowed dumping the flash.

**Comment by jclehner** [comment]
Sorry, forgot about that. At the minimum, please add a `"flash"` section under `spaces`, including a `blocksize` of 64k.

A partition map, and the `.blsig` would be nice, but we can add that later. Can you send me a dump of the bootloader? Or upload it to https://github.com/jclehner/bcm2-dumps?

**Comment by redawl** [comment]
> [screenlog.txt](https://github.com/jclehner/bcm2-utils/files/14932587/screenlog.txt) [bootloader2.txt](https://github.com/jclehner/bcm2-utils/files/14932591/bootloader2.txt)
> 
> Interesting that deviwiki says 3384, while the bootlog shows 3843. Maybe a typo in the firmware? I will fix :)
> 
> flashchip is [25Q64FVAIG](https://pdf1.alldatasheet.com/datasheet-pdf/view/1156000/WINBOND/25Q64FVAIG.html)

I attatched bootloader and boot logs to this message here. 

Partition map doesn't work because the bootloader menu does not have a 'p' command to print the partition table. I believe I had the same issue with flash space, but I can retest.

**Comment by redawl** [comment]
PR with flash dump and bootloader dump from cm500:
https://github.com/jclehner/bcm2-dumps/pull/4

**Comment by jclehner** [comment]
Please add a `.blsig` of `0x3384`, and the partition layout I've posted in [your PR](https://github.com/jclehner/bcm2-dumps/pull/4) in bcm2-dumps.

Then it's ready to be merged.

**Comment by jclehner** [comment]
> Dumping via flash does not work however, since there is no 'p' command as discussed above in my first comment.

That command is only required to get an idea of the partition layout, but we've figured that out already.

Dumping flash from the bootloader identification of the bootloader's `printf` function, plus a function that reads from flash into a specified buffer. See [this example](https://github.com/jclehner/bcm2-utils/blob/8ec73865655eabdafbcd2655419de30b7435f958/profiledef.c#L179).

> Let me know what else is needed for this to get merged

**Pull request merged**